tempodb/encoding/common/interfaces.go

```diff
@@ -32,6 +32,11 @@ type Searcher interface {
 	FetchTagNames(context.Context, traceql.FetchTagsRequest, traceql.FetchTagsCallback, SearchOptions) error
 }
 
+type BlockCacher interface {
```

Comment:
I'm ok with leaving this out until we cache items in non-LocalBlocks.   And make these methods part of `LocalBlock`. If we split `searchBlock` like I mentioned above, the caching variant can take the `*LocalBlock` directly.

Reply:
cool, can LocalBlock and keep it simple